Raven.Database/Storage/Esent/StorageActions/Documents.cs

```diff
@@ -337,6 +354,12 @@ private JsonDocument ReadCurrentDocument()
 
             var errors = new List<DocumentFetchError>();
             var skipDocumentGetErrors = failedToGetHandler != null;
+            var hasEntityNames = entityNames != null && entityNames.Count > 0;
+            if (hasEntityNames)
```

Comment:
Here you convert it to case insensitive, but you are  doing other stuff with it elsewhere that are case sensitive.